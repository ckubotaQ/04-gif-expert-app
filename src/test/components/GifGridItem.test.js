import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem>', () => {
    const title= 'Prueba Bojack'; 
        const url= 'https://localhost:/alo.jpg'
    const wrapper = shallow( <GifGridItem title={title} url={url}/> );
    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de tener un parrafo con el title ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
        
    });
    test('debe de obtener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        //console.log(img.props('src'))
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    test('debe tener la clase animate__bounce', () => {
        const div = wrapper.find('div');
        const classname =div.prop('className');
        expect(classname.includes('animate__bounce')).toBe(true);
    })
    
    
})
