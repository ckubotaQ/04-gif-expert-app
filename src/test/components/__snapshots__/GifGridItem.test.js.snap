// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Prueba Bojack"
    src="https://localhost:/alo.jpg"
  />
  <p>
    Prueba Bojack
  </p>
</div>
`;
